File: data/transcoder_evaluation_gfg/c/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

int f_gold ( char str [] ) {
  int checker = 0;
  for ( int i = 0;
  i < strlen(str);
  ++ i ) {
    int val = ( str [ i ] - 'a' );
    if ( ( checker & ( 1 << val ) ) > 0 ) return i;
    checker |= ( 1 << val );
  }
  return - 1;
}


int f_filled ( char str [] ) {}

int main(void) {
    int n_success = 0;
    char param0[][100] = {"XFGfXTDgpIuerN","5621946166","11010110","xL","2575","0100010","SZmmQ","9735892999350","1001101101101","oEXDbOU"};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/go/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(str []byte) int {
	var checker int = 0
	for i := int(0); i < int(len(str)); i++ {
		var val int = int(str[i] - byte('a'))
		if (checker & (1 << val)) > 0 {
			return i
		}
		checker |= 1 << val
	}
	return -1
}
//TOFILL
func main() {
	var (
		n_success int           = 0
		param0    [10][]byte = [10][]byte{func() []byte {
			var t []byte
			copy(t[:], ([]byte)("XFGfXTDgpIuerN"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("5621946166"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("11010110"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("xL"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("2575"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("0100010"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("SZmmQ"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("9735892999350"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("1001101101101"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("oEXDbOU"))
			return t
		}()}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i][:]) == f_gold(param0[i][:]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}
